src/components/TransactionImport/Form.tsx

```diff
@@ -0,0 +1,119 @@
+import { useTranslation } from 'react-i18next'
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { Account, TransactionPreview, Translation } from '../../types'
+import { parseFile } from '@envelope-zero/ynap-parsers'
+import { checkStatus, parseJSON } from '../../lib/fetch-helper'
+import Error from '../Error'
+import FormFields from '../FormFields'
+import FormField from '../FormField'
+import Autocomplete from '../Autocomplete'
+import LoadingSpinner from '../LoadingSpinner'
+import { safeName } from '../../lib/name-helper'
+
+type Props = {
+  accounts: Account[]
+  isLoading: boolean
+  setIsLoading: (isLoading: boolean) => void
+  setResult: (result: TransactionPreview[], targetAccountId: string) => void
+}
+
+const Form = ({ accounts, isLoading, setIsLoading, setResult }: Props) => {
+  const { t }: Translation = useTranslation()
+  const [error, setError] = useState('')
+  const [accountId, setAccountId] = useState('')
+
+  return (
+    <form
+      encType="multipart/form-data"
+      onSubmit={event => {
+        event.preventDefault()
+        const file = (event.target as any).file.files[0] // TODO: without `any`
+
+        parseFile(file)
+          .then(result => {
+            // Get the data and transform it to FormData for the backend
+            const data = new FormData()
+            data.append(
+              'file',
+              new File([new Blob([result[0].data])], 'ynap-transactions.csv', {
+                type: 'text/csv;charset=utf-8;',
+              })
+            )
```

Comment:
This takes the result of the YNAP parsing and transforms it into data that can be uploaded to the transaction preview endpoint.